src/components/MdeToolbar.tsx

```diff
@@ -8,45 +8,85 @@ export interface MdeToolbarProps {
     commands: Command[][];
     onCommand: (command: Command) => void;
     readOnly: boolean;
+    sticky: boolean;
 }
 
-export const MdeToolbar: React.SFC<MdeToolbarProps> = ({children, commands, onCommand, readOnly}) => {
-    if ((!commands || commands.length === 0) && !children) {
-        return null;
+export interface MdeToolbarState {
+    scrolled: boolean;
+    sticky: boolean;
+    scrollHeight?: number;
+}
+
+export class MdeToolbar extends React.Component<MdeToolbarProps, MdeToolbarState> {
+    constructor(props: MdeToolbarProps) {
+        super(props);
+        this.state = {
+            scrolled: false,
+            sticky: props.sticky,
+            scrollHeight: null,
+        };
+        this.handleOnScroll = this.handleOnScroll.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.sticky) {
+            window.addEventListener("scroll", this.handleOnScroll);
+            window.addEventListener("touchend", this.handleOnScroll);
+        }
+    }
+
+    handleOnScroll() {
+        const scrollTop = window.scrollY;
+        const toolbars = document.getElementsByClassName("mde-header") as HTMLCollectionOf<HTMLElement>;
+        const toolbarScroll = toolbars[0].offsetTop;
+        if (!this.state.scrollHeight) this.setState({ scrollHeight: toolbarScroll });
+        if (scrollTop > toolbarScroll && this.state.scrolled === false && this.props.sticky) {
```

Comment:
Not of a fan of mixing setState and scrolling as I've had performance issues before 

https://www.w3schools.com/howto/howto_js_sticky_header.asp
Coulda also done this

Also you forgot to set this.scrollListenerId + this.touchendListenerId and remove them via componentWillUnmount()

Reply:
Implementation's fine though if it works, only a sticky bar tbh
Just add the componentWillUnmount

Reply:
Looking forward to useEffect hooks as its api is concise for exactly this!

Reply:
Done :D